### PR TITLE
add find! method to ActiveQuery

### LINKED ISSUE #38: Add `find!` method
It should behave like `find`, except it raises an `ActiveForce::RecordNotFound` if no record is found.

### PR DESCRIPTION
**Pull request by rferg**
Closes #38

~~There were some extraneous formatting changes that were accidentally included.  Let me know if you want to revert those.~~

### CODE REVIEW

File: lib/active_force/sobject.rb [hunk 1]
```diff
@@ -27,7 +27,8 @@ class SObject
 
     class << self
       extend Forwardable
-      def_delegators :query, :not, :or, :where, :first, :last, :all, :find, :find_by, :find_by!, :sum, :count, :includes, :limit, :order, :select, :none
+      def_delegators :query, :not, :or, :where, :first, :last, :all, :find, :find!, :find_by, :find_by!, :sum, :count,
```
Comment (rferg) [inline lib/active_force/sobject.rb:30]: Auto-formatter accidentally ran on save for this file.  Only real change is adding `:find!` to list of delegated methods.

File: spec/active_force/active_query_spec.rb [hunk 2]
```diff
@@ -188,77 +191,118 @@
...
+    it 'raises if record not found' do
       allow(client).to receive(:query).and_return(build_restforce_collection)
-      expect { active_query.find_by!(field: 123) }.to raise_error(ActiveForce::RecordNotFound)
+      expect { active_query.find_by!(field: 123) }
+        .to raise_error(ActiveForce::RecordNotFound, "Couldn't find #{sobject.table_name} with {:field=>123}")
     end
   end
 
-  describe "responding as an enumerable" do
+  describe '#find!' do
```
Comment (rferg) [inline spec/active_force/active_query_spec.rb:237]: Same on this file.  I changed the test above for `find_by!` to check the exception message and added the tests for `find!`.

**Comment by asedge** [comment]
@rferg Could you revert the formatting changes, please?

I do think we should probably add Rubocop to this repo at some point and when we do we can do a mass linter autofix.

**Comment by rferg** [comment]
> @rferg Could you revert the formatting changes, please?
> 
> I do think we should probably add Rubocop to this repo at some point and when we do we can do a mass linter autofix.

@asedge Formatting reverted.

**Review by asedge — APPROVED** [review]
I love it, thanks so much for adding this!

**Pull request merged**